src/infra/ssl/sslCmdRepo.go

```diff
@@ -173,75 +180,119 @@ func (repo *SslCmdRepo) shouldIncludeWww(vhost valueObject.Fqdn) bool {
 	return false
 }
 
-func (repo *SslCmdRepo) ReplaceWithValidSsl(sslPair entity.SslPair) error {
-	path, _ := valueObject.NewMappingPath(infraData.GlobalConfigs.DomainOwnershipValidationUrlPath)
-	matchPattern, _ := valueObject.NewMappingMatchPattern("equals")
-	targetType, _ := valueObject.NewMappingTargetType("inline-html")
-	httpResponseCode, _ := valueObject.NewHttpResponseCode(200)
+func (repo *SslCmdRepo) filterDomainsMappedToServer(
+	vhostNames []valueObject.Fqdn,
+	expectedOwnershipHash valueObject.Hash,
+) []valueObject.Fqdn {
+	domainsMappedToServer := []valueObject.Fqdn{}
 
-	expectedOwnershipHash, err := repo.sslQueryRepo.GetOwnershipValidationHash(
-		sslPair.Certificate.CertificateContent,
-	)
-	if err != nil {
-		return errors.New("CreateOwnershipValidationHashError: " + err.Error())
-	}
-	targetValue, _ := valueObject.NewMappingTargetValue(
-		expectedOwnershipHash.String(), targetType,
-	)
+	for _, vhostName := range vhostNames {
+		vhostNameStr := vhostName.String()
+		vhostNameIps, err := net.LookupIP(vhostNameStr)
+		if err != nil || len(vhostNameIps) == 0 {
+			continue
+		}
 
-	firstVhostName := sslPair.VirtualHostsHostnames[0]
-	inlineHtmlMapping := dto.NewCreateMapping(
-		firstVhostName,
-		path,
-		matchPattern,
-		targetType,
-		&targetValue,
-		&httpResponseCode,
-	)
+		var serverIpAddress *valueObject.IpAddress
+		for _, vhostNameIp := range vhostNameIps {
+			ipAddress, err := valueObject.NewIpAddress(vhostNameIp.String())
+			if err != nil {
+				continue
+			}
 
-	mappingCmdRepo := mappingInfra.NewMappingCmdRepo(repo.persistentDbSvc)
-	mappingId, err := mappingCmdRepo.Create(inlineHtmlMapping)
-	if err != nil {
-		return errors.New("CreateOwnershipValidationMappingError: " + err.Error())
-	}
+			serverIpAddress = &ipAddress
+			break
+		}
 
-	isDomainMappedToServer := repo.isDomainMappedToServer(
-		firstVhostName,
-		expectedOwnershipHash,
-	)
+		if serverIpAddress == nil {
+			continue
+		}
 
-	mappingQueryRepo := mappingInfra.NewMappingQueryRepo(repo.persistentDbSvc)
-	mappings, err := mappingQueryRepo.ReadByHostname(firstVhostName)
-	if err != nil {
-		return errors.New("ReadVhostMappingsError: " + err.Error())
+		ownershipValidationMappingId, err := repo.createOwnershipValidationMapping(
+			vhostName,
+			expectedOwnershipHash,
+		)
+		if err != nil {
+			continue
+		}
+
+		hashUrlPath := infraData.GlobalConfigs.DomainOwnershipValidationUrlPath
+		hashUrlFull := "https://" + vhostNameStr + hashUrlPath
+		serverIpStr := serverIpAddress.String()
+
+		ownershipHashFound, err := infraHelper.RunCmd(
+			"curl", "-skLm", "10", "--resolve", vhostNameStr+":443:"+serverIpStr, hashUrlFull,
+		)
+		if err != nil {
+			hashUrlFull = "https://" + serverIpStr + hashUrlPath
+			ownershipHashFound, err = infraHelper.RunCmd(
+				"curl", "-skLm", "10", "-H", "Host: "+vhostNameStr, hashUrlFull,
+			)
+			if err != nil {
+				continue
```

Comment:
Hmmm, continue, is this a for loop? I have the impression that the actual ownership validation via HTTP should be the method itself while the for loop is on the main method, let's see.